### PR TITLE
Cross platform acceptance tests

### PR DESCRIPTION
**Pull request by nohwnd**
## Description
Run most of the acceptance tests on Linux and MacOS.

### CODE REVIEW

File: test/Microsoft.TestPlatform.AcceptanceTests/BlameDataCollectorTests.cs [hunk 1]
```diff
@@ -158,9 +171,9 @@ public void CrashDumpOnExit(RunnerInfo runnerInfo)
         }
 
         [TestMethod]
-        [NetFrameworkRunner("net452;net472;netcoreapp3.1;net5.0")]
+        [NetCoreRunner("net452;net472;netcoreapp3.1;net5.0")]
```
Comment (Sanan07) [inline test/Microsoft.TestPlatform.AcceptanceTests/BlameDataCollectorTests.cs:174]: What is the reason of changing attributes order?

File: test/Microsoft.TestPlatform.AcceptanceTests/Extension/NetCoreRunner.cs [hunk 2]
```diff
@@ -24,7 +25,10 @@ public class NetCoreRunner : Attribute, ITestDataSource
         /// <param name="targetFrameworks">To run tests with desktop runner(vstest.console.exe), use AcceptanceTestBase.Net452TargetFramework or alike values.</param>
         public NetCoreRunner(string targetFrameworks = AcceptanceTestBase.NETFX452_NET50)
         {
-            foreach (var fmw in targetFrameworks.Split(';'))
+            var isWindows = System.Environment.OSVersion.Platform.ToString().StartsWith("Win");
```
Comment (Sanan07) [inline test/Microsoft.TestPlatform.AcceptanceTests/Extension/NetCoreRunner.cs:28]:
```suggestion
            var isWindows = Environment.OSVersion.Platform.ToString().StartsWith("Win");
```

File: test/Microsoft.TestPlatform.AcceptanceTests/Extension/NetFrameworkRunner.cs [hunk 3]
```diff
@@ -24,6 +24,12 @@ public class NetFrameworkRunner : Attribute, ITestDataSource
         /// <param name="targetFrameworks">To run tests with desktop runner(vstest.console.exe), use AcceptanceTestBase.Net452TargetFramework or alike values.</param>
         public NetFrameworkRunner(string targetFrameworks = AcceptanceTestBase.NETFX452_NET50)
         {
+            var isWindows = System.Environment.OSVersion.Platform.ToString().StartsWith("Win");
```
Comment (Sanan07) [inline test/Microsoft.TestPlatform.AcceptanceTests/Extension/NetFrameworkRunner.cs:27]: The same as above. Simplify namespace

File: test/Microsoft.TestPlatform.AcceptanceTests/DifferentTestFrameworkSimpleTests.cs [hunk 4]
```diff
@@ -67,6 +70,7 @@ public void WebTestRunAllTestsWithRunSettings(RunnerInfo runnerInfo)
         }
 
         [TestMethod]
+        [TestCategory("Windows-Review")]
```
Comment (Sanan07) [inline test/Microsoft.TestPlatform.AcceptanceTests/DifferentTestFrameworkSimpleTests.cs:73]: What is difference between 'Windows' and 'Windows-Review' test category?

File: test/Microsoft.TestPlatform.AcceptanceTests/BlameDataCollectorTests.cs (hunk 1 above)
Comment (nohwnd) [reply test/Microsoft.TestPlatform.AcceptanceTests/BlameDataCollectorTests.cs:174]: .net framework does not run on Linux, unless we want to test it on mono

File: test/Microsoft.TestPlatform.AcceptanceTests/DifferentTestFrameworkSimpleTests.cs (hunk 4 above)
Comment (nohwnd) [reply test/Microsoft.TestPlatform.AcceptanceTests/DifferentTestFrameworkSimpleTests.cs:73]: Windows-Review is added in this PR, those are tests that run currently only on windows but should be reviewed again to make sure we are running all tests that can run on Linux.

File: test/Microsoft.TestPlatform.AcceptanceTests/Extension/NetCoreTargetFrameworkDataSource.cs [hunk 5]
```diff
@@ -34,7 +34,8 @@ public class NetCoreTargetFrameworkDataSource : Attribute, ITestDataSource
             // all tests to avoid changing all acceptance tests right now
             bool useNetCore31Target = false)
         {
-            if (useDesktopRunner)
+            var isWindows = System.Environment.OSVersion.Platform.ToString().StartsWith("Win");
```
Comment (nohwnd) [inline test/Microsoft.TestPlatform.AcceptanceTests/Extension/NetCoreTargetFrameworkDataSource.cs:37]:
```suggestion
            var isWindows = Environment.OSVersion.Platform.ToString().StartsWith("Win");
```

File: test/Microsoft.TestPlatform.AcceptanceTests/Extension/NetFrameworkRunner.cs (hunk 3 above)
Comment (nohwnd) [reply test/Microsoft.TestPlatform.AcceptanceTests/Extension/NetFrameworkRunner.cs:27]:
```suggestion
            var isWindows = Environment.OSVersion.Platform.ToString().StartsWith("Win");
```

File: test/Microsoft.TestPlatform.AcceptanceTests/Extension/NetFullTargetFrameworkDataSource.cs [hunk 6]
```diff
@@ -29,12 +29,13 @@ public NetFullTargetFrameworkDataSource(bool inIsolation = true, bool inProcess
         {
             this.dataRows = new List<object[]>();
 
-            if (useCoreRunner)
+            var isWindows = System.Environment.OSVersion.Platform.ToString().StartsWith("Win");
```
Comment (nohwnd) [inline test/Microsoft.TestPlatform.AcceptanceTests/Extension/NetFullTargetFrameworkDataSource.cs:32]:
```suggestion
            var isWindows = Environment.OSVersion.Platform.ToString().StartsWith("Win");
```

**Review by Sanan07 — APPROVED** [review]

**Review by Haplois — APPROVED** [review]
LGTM

**Pull request merged**